db/python/layers/sequencing_group.py

```diff
@@ -347,7 +347,7 @@ async def upsert_sequencing_groups(
         # but we're inside a transaction, so it's not actually committing anything
         # so should be quick to "write" in serial
         for sg in to_insert:
-            assay_ids = [a.id for a in sg.assays]
+            assay_ids = [a.id for a in sg.assays] if sg.assays else None
```

Comment:
The table `create_sequencing_group` method is expecting assay_ids to be of type `list[str]`, so we shouldn't pass None here. The check in the table method will work with an empty list too, so you should be able to replace `None` with `[]` here.